kv/dist_sender.go

```diff
@@ -255,7 +255,8 @@ func (ds *DistSender) optimizeReplicaOrder(replicas replicaSlice) rpc.OrderingPo
 	}
 	// Sort replicas by attribute affinity, which we treat as a stand-in for
 	// proximity (for now).
-	if replicas.SortByCommonAttributePrefix(nodeDesc.Attrs.Attrs) > 0 {
+	len := replicas.SortByCommonAttributePrefix(nodeDesc.Attrs.Attrs)
```

Comment:
Shadowing the global function `len()` is a bad idea. Put this back the way it was, and if you do need a variable for something like this I suggest a name like `commonPrefixLen`.